### PR TITLE
docs: Add Running BDD tests in detached mode

### PR DESCRIPTION
**Pull request by Baha-sk**
part of # 739

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ The packages intended for end developer usage are within the [pkg/client](pkg/cl
 ## Testing
 - [Build](docs/test/build_instructions.md)
 - [BDD tests](docs/test/bdd_instructions.md)
+- [BDD tests in detached mode](docs/test/starting_tests_fixtures.md)
```
Comment (troyronda) [inline README.md]: Why not have this as part of the BDD tests markdown file?

Comment (Baha-sk) [reply README.md]: to separate detached mode instructions from normal bdd tests instruction. It's easier to read this way.. Should I merge instead?

File: docs/test/starting_tests_fixtures.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+via the make target mentioned in [bddtests_instruction.md](bddtests_instruction.md), then do the following:
+
+1. Navigate to test/bdd/fixtures
+
+2. To start Alice and Bob agents and webhooks, go to agents folder and run:
+```shell script
+(source .env && docker-compose down && NO_PROXY=* docker-compose up --force-recreate)
+```
+   
+3. To start the Demo API hosts for Alic and Bob, open up a new terminal and go demo/openapi folder and run the
```
Comment (rolsonquadras) [inline docs/test/starting_tests_fixtures.md]: typo - Alice

File: README.md (hunk 1 above)
Comment (troyronda) [reply README.md]: Or you could put this link into that file.

@rolsonquadras  thoughts?

Comment (rolsonquadras) [reply README.md]: Agree with @troyronda ie putting link for the new file in BDD tests. I see it as a part of BDD tests.

Comment (Baha-sk) [reply README.md]: done

File: docs/test/starting_tests_fixtures.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ ```shell script
+ DISABLE_COMPOSITION=true go test -run didexchange_e2e_sdk -count=1 -v -cover . -p 1 -timeout=20m -race
+ ```
+or
+ ```shell script
+ DISABLE_COMPOSITION=true go test -run all -count=1 -v -cover . -p 1 -timeout=20m -race
+ ``` 
+
+using `-run all` is the equivalent of not specifying the run argument.
+At the time of writing this doc, the following are available run targets, they are located in the .feature files:
```
Comment (rolsonquadras) [inline docs/test/starting_tests_fixtures.md]: Looks like we are duplicating the content - https://github.com/hyperledger/aries-framework-go/blob/master/docs/test/bdd_tests.md#demonstration.

File: docs/test/starting_tests_fixtures.md (hunk 2 above)
Comment (Baha-sk) [reply docs/test/starting_tests_fixtures.md]: done

File: docs/test/build.md [hunk 4]
```diff
@@ -24,6 +24,9 @@ make unit-test
 make bdd-test
 ```
 
+## Running BDD tests in detached mode
```
Comment (rolsonquadras) [inline docs/test/build.md]: Feels like these lines should be added in [bdd_test.md](https://github.com/hyperledger/aries-framework-go/blob/master/docs/test/bdd_tests.md#run-bdd-tests) file.

File: docs/test/starting_tests_fixtures.md (hunk 3 above)
Comment (Baha-sk) [reply docs/test/starting_tests_fixtures.md]: this is the list of runnable features.. the demo is probably making use of these, but the list is not meant to be exhaustive as new feature will be added in the future (I doubt this readme will be updated then :))

File: docs/test/build.md (hunk 4 above)
Comment (Baha-sk) [reply docs/test/build.md]: ok..

Comment (Baha-sk) [reply docs/test/build.md]: done

File: docs/test/starting_tests_fixtures.md (hunk 3 above)
Comment (rolsonquadras) [reply docs/test/starting_tests_fixtures.md]: yeah, but may be pointing to one instance(either this or on the page shared on first link) would be helpful. Duplicates are hard to maintain and quickly goes out of sync.

Comment (Baha-sk) [reply docs/test/starting_tests_fixtures.md]: the intent was not to maintain the list.. just to show examples.. and the reader can figure out other features by reading the files.

File: docs/test/starting_tests_fixtures.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+2. To start Alice and Bob agents and webhooks, go to agents folder and run:
+```shell script
+(source .env && docker-compose down && NO_PROXY=* docker-compose up --force-recreate)
+```
+   
+3. To start the Demo API hosts for Alice and Bob, open up a new terminal and go demo/openapi folder and run the 
+above command.
+
+4. To start SideTree Mock service, open up a new terminal go to sidetree-mock folder and run the same above command.
```
Comment (troyronda) [inline docs/test/starting_tests_fixtures.md]: Sidetree

Comment (Baha-sk) [reply docs/test/starting_tests_fixtures.md]: done

**Comment by fqutishat** [comment]
duplicate documentation similar to https://github.com/hyperledger/aries-framework-go/blob/master/docs/test/bdd_tests.md#demonstration.

File: docs/test/starting_tests_fixtures.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+```
+   
+3. To start the Demo API hosts for Alice and Bob, open up a new terminal and go demo/openapi folder and run the 
+above command.
+
+4. To start Sidetree Mock service, open up a new terminal go to sidetree-mock folder and run the same above command.
+
+5. Once all the containers above have started, you are ready to run BDD tests manually.
+
+6. In a new terminal, navigate to tests/bdd folder and run the same test command found in scripts/checke_integration.sh
```
Comment (troyronda) [inline docs/test/starting_tests_fixtures.md]: typo

Comment (Baha-sk) [reply docs/test/starting_tests_fixtures.md]: updated

**Comment by Baha-sk** [comment]
> duplicate documentation similar to https://github.com/hyperledger/aries-framework-go/blob/master/docs/test/bdd_tests.md#demonstration.

This documentation focuses on running tests in detached mode (regardless of the demo scenarios), while demonstration describes the demo scenarios included in the integration tests. Both are useful to the reader.

**Comment by Baha-sk** [comment]
> > duplicate documentation similar to https://github.com/hyperledger/aries-framework-go/blob/master/docs/test/bdd_tests.md#demonstration.
> 
> This documentation focuses on running tests in detached mode (regardless of the demo scenarios), while demonstration describes the demo scenarios included in the integration tests. Both are useful to the reader.

new file deleted .. and moved doc to add_tests.md

File: docs/test/starting_tests_fixtures.md (hunk 3 above)
Comment (Baha-sk) [reply docs/test/starting_tests_fixtures.md]: removed list in the new doc now.

**Pull request merged**